docs: update Algolia docs for new DocSearch infra (#6393)
facebook/docusaurus: website/docs/search.md

> The service is **free** in most cases: just [apply to the DocSearch program](https://docsearch.algolia.com/apply).

Reviewer: ```suggestion
The service is **free** for any open-source projects: just make sure to read the [checklist](https://docsearch.algolia.com/docs/who-can-apply/) and [apply to the DocSearch program](https://docsearch.algolia.com/apply).
```

(A lot of people are applying for commercial projects)

> It works by regularly crawling the content of your website (the schedule is configurable from the web interface) and putting all the content in an Algolia index. This content is then queried directly from your front-end using the Algolia API.

Reviewer: ```suggestion
It crawls your website once a week (the schedule is configurable from the web interface) and index all the content in an Algolia index. This content is then queried directly from your front-end using the Algolia API.
```

A bit repetitive with the `index` word but I feel like this is more appropriate, since they will see those terms quite often while browsing our interfaces

:::

:::info From the old docsearch?

> You can read more about migration from the legacy DocSearch infra in [our blog post](/blog/2021/11/21/algolia-docsearch-migration).

Reviewer: I know it's already linked in the blog post, but we also have this migration guide here: https://docsearch.algolia.com/docs/migrating-from-legacy

:::

:::info From an old docsearch version?

> You can read more about migration from DocSearch v2 and configurating v3 in [our blog post](/blog/2021/11/21/algolia-docsearch-migration).

Reviewer: Actually v2/v3 is the UI, we don't have versioning for the infra, it's just `legacy` for the previous one.

We can remove the versions since Docusaurus v2 only support DocSearch v3 UI components

:::

### Index Configuration {#algolia-index-configuration}

> After your application has been approved, you can view and update your search configuration through the web interface.

Reviewer: ```suggestion
After your application has been approved and deployed, you will receive an email with all the details for you to add DocSearch to your project. Editing and managing your crawls can be done via [the web interface](https://crawler.algolia.com/)
```

It's maybe worth mentioning that indices are ready to use when deployed, which means configuration is not necessary, wdyt?
